File: VB/DrillDownWebReport/XtraReport1.vb
Imports System.Drawing
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI

Namespace DrillDownWebReport

    Public Partial Class XtraReport1
        Inherits XtraReport

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub lbShowHide_BeforePrint(ByVal sender As Object, ByVal e As CancelEventArgs)
            lbShowHide.Text = If(DetailReport.DrillDownExpanded, "Hide Details", "Show Details")
        End Sub
    End Class
End Namespace

File: CS/DrillDownWebReport/XtraReport1.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;

namespace DrillDownWebReport {
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport {
        public XtraReport1() {
            InitializeComponent();
        }

        private void lbShowHide_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e) {
            lbShowHide.Text = DetailReport.DrillDownExpanded ? "Hide Details" : "Show Details";
        }

    }
}
